source-git-monorepos/README.md

```diff
@@ -50,21 +89,367 @@ packages:
 
 With the above source-git configuration could be kept in
 `.distro/source-git.yaml`. Downstream files to be synced to dist-git are to be
-kept in subdirectories, named according to the distro package, in `.distro` to
-keep things organized.
+kept in subdirectories, named according to `package_name` (the key used in
+`source-git.yaml` to reference the package), in `.distro`, to keep things
+organized.
+
+An example layout:
+
+```
+.distro
+├── backend
+│   └── copr-backend.spec
+├── cli
+│   └── copr-cli.spec
+├── frontend
+│   └── copr-frontend.spec
+└── source-git.yaml
+```
 
 This way all distribution configs and files would be kept at one place, which
-probably is somewhat better than having source-git configuration spread
-across the monorepo, in subdirectories. This setup also allows multiple paths
-from the repo mapped to a dist-git repo.
+probably is somewhat better than having source-git configuration spread across
+the monorepo, in subdirectories. This setup also allows multiple paths from
+the repo mapped to a dist-git repo.
+
+### `packit source-git init`
+
+If the source-git repository to be initialized is a monorepo, then each
+dist-git repository specified will also need to one or more subdirectories of
+the source-git repository.
+
+Which makes the whole thing a little bit too complicated ☺
+
+This complication could be elevated, if `packit source-git init` would use a
+guided (dialog-like) process, instead of taking all its inputs as arguments.
+
+As an alternative, the command could have a variant where it takes a basic
+`packages` definition to guide the initialization of a source-git monorepo.
+
+Example `packages.yaml`:
+
+```yaml
+- downstream_package_name: copr-backend
+  package_path: backend
+  dist_git_path: rpms/copr-backend
+  dist_git_branch: rawhide
+- downstream_package_name: copr-frontend
+  package_path: frontend
+  dist_git_path: rpms/copr-frontend
+  dist_git_branch: rawhide
+- downstream_package_name: copr-cli
+  package_path: cli
+  dist_git_path: rpms/copr-cli
+  dist_git_branch: rawhide
```

Comment:
I bet only Fedora dist-git is supported, now or forever?  Copr identifies also CentOS, CentOS Stream and Red Hat (internally). Plus four Copr dist-git instances, but these are not open for PRs.  So just a feedback, DistGit operations in Copr have the `--distgit <DISTGIT_ID` option (which hides the dist-git instance configuration details).

Reply:
All dist-gits are supported, I just tend to use Fedora in examples.

As of now mapping the source-git repo to a dist-git instance is not configurable for the users only for the bots, as we expect CentOS to want to have their own source-git repos in their own place (which are mapped to CentOS dist-git) and Fedora to want their own source-git repos in their own place, mapped to Fedora dist-git repos.